test/typedarray/TypedArrayBuiltins.js

```diff
@@ -83,6 +83,25 @@ var tests = [
                 assert.areEqual(2, arr[1], "TypedArray " + t + " created from iterable has element #1 == 2");
                 assert.areEqual(3, arr[2], "TypedArray " + t + " created from iterable has element #2 == 3");
             }
+
+            for(var t of TypedArray) {
+                var a = [1,2,3,4];
+                a[Symbol.iterator] = getIterableObj([99,0])[Symbol.iterator];
+                var arr = new this[t](a);
+                assert.areEqual(1, arr.length, "TypedArray " + t + " created from array with user-defined iterator has length == 1");
+                assert.areEqual(99, arr[0], "TypedArray " + t + " created from array with user-defined iterator has element #0 == 99");
+            }
+
+            var builtinArrayPrototypeIteratorDesc = Object.getOwnPropertyDescriptor(Array.prototype, Symbol.iterator);
+            for(var t of TypedArray) {
+                var a = [1,2,3,4];
+                Object.defineProperty(Array.prototype, Symbol.iterator,
+                    {enumerable: false, configurable: true, writable: true, value:getIterableObj([99,0])[Symbol.iterator]});

```

Comment:
This doesn't need to be inside the loop.  Or resetting it should also be inside the loop.